doc/nrf/gs_testing.rst

```diff
@@ -120,3 +120,93 @@ To connect to the nRF9160-based kit with LTE Link Monitor, perform the following
+
+If you use a multi-core SoC, for example from the nRF53 Series, you need a single debug session to debug the application core firmware.
+To debug the firmware running on the network core, you need to set up two separate debug sessions: one for the network core and one for the application core.
+When debugging the network core, the application core debug session runs in the background and you can debug both cores if needed.
+
+Complete the following steps to start debugging the network core:
+
+1. Set up sessions for the application core and network core as mentioned in `Debugging an application`_.
+#. Select the appropriate CPU for debugging in each session, corresponding to the application core and the network core of your SoC, respectively.
+#. Once both sessions are established, execute the code on the application core.
+
+   The startup code releases the ``NETWORK.FORCEOFF`` signal to start the network core and allocates the necessary GPIO pins for it.
+#. Start code execution on the network core in the other debug session.
+
+If you want to reset the network core while debugging, make sure to first reset the application core and execute the code.
+
+.. note::
+  Debugging firmware on the application core in the non-secure domain (that is, when using an ``_ns`` build target), is currently not supported.
+
+Debug configuration
+===================
+
+The following Kconfig options are used by the |VSC| debugger in the |nRFVSC|:
+
+* :kconfig:option:`CONFIG_DEBUG_OPTIMIZATIONS` set to ``y``
+* :kconfig:option:`CONFIG_DEBUG_THREAD_INFO` set to ``y``
+
+There are many more Kconfig options for debugging that are specific to different modules.
+For details, see the respective documentation pages of the modules.
+
+Debug build types
+-----------------
+
+Some applications and samples provide a specific build type that enables additional debug functionalities.
+You can select build types when you are :ref:`configuring the build settings <gs_modifying_build_types>`.
+
+Debug logging
+-------------
+
+You can use logging system to get more information about the state of your application.
+Logs are integrated to many various modules and subsystems in the |NCS| and Zephyr.
+These logs are visible once you configure logger for your application.
+
+You can also configure log level per logger module, for example to get more information about a given subsystem.
+See :ref:`ug_logging` for details on how to enable and configure logs.
+
+Debug libraries
+---------------
+
+The |NCS| also provides several libraries and drivers for debugging different components.
+For example:
+
+* You can use :ref:`lib_debug` for any of your applications, for example to measure CPU load or trace hardware peripheral events on pins.
+
+* You can use :ref:`nrf_profiler` to measure performance and debug applications without introducing big performance overhead.
+  This option requires introducing additional code changes to your application.
```

Comment:
Maybe we could add that:
`Your application must register profiler events and log their occurrences.`

Reply:
To clarify what is needed